src/jsx.d.ts

```diff
@@ -0,0 +1,9 @@
+import {VNode, VNodeData} from './vnode';
+
+declare global {
+  namespace JSX {
+    interface IntrinsicElements {
+      [elemName: string]: VNodeData;
+    }
+  }
+}
```

Comment:
I've left it a bit free because I don't want snabbdom to be tied to html spec and what elements are accetable / not acceptable. Especially in world of custom elements where a user can create any name, we don't want things too strict. The user can make them strict if they wish.

e.g

```
<my-tag attrs={{super: true}} on={{superEvent: this.handleSuperEvent}} />
```

can be typed in the user library as.

```
interface MyTagData extends VNodeData {
  attrs: {
    super: boolean;
  }
  on: {
   superEvent: EventListener
  }
}

declare global {
  namespace JSX {
    interface IntrinsicElements {
      'my-tag': MyTagData;
    }
  }
}
```

since TS merges interfaces, this will support both my-tag and other html elements. For my-tag the compiler will be extra strict and validate that only certain attributes and events are allowed. It will also provide really nice code completion when coding jsx because it can infer the structure from MyTagData.